azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py

```diff
@@ -0,0 +1,335 @@
+        if not isinstance(original_aep, dict):
+            original_aep = original_aep.as_dict()
+        return original_aep["properties"]["transportAuthentication"]
+
+
+# Helpers
+def _process_authentication(
+    auth_mode: Optional[str] = None,
+    auth_props: Optional[Dict[str, str]] = None,
+    certificate_reference: Optional[str] = None,
+    password: Optional[str] = None,
+    username: Optional[str] = None
+) -> Dict[str, str]:
+    if not auth_props:
+        auth_props = {}
+    # add checking for ensuring auth mode is set with proper params
+    if certificate_reference and (username or password):
+        raise Exception("Please choose to use a certificate reference or a username/password for authentication.")
+
+    if certificate_reference and auth_mode in [None, AEPAuthModes.certificate.value]:
+        auth_props["mode"] = AEPAuthModes.certificate.value
+        auth_props["x509Credentials"] = {"certificateReference": certificate_reference}
+        if auth_props.pop("usernamePasswordCredentials", None):
+            logger.warning("Previously used username and password references were removed.")
+
+    elif (username or password) and auth_mode in [None, AEPAuthModes.userpass.value]:
+        auth_props["mode"] = AEPAuthModes.userpass.value
+        user_creds = auth_props.get("usernamePasswordCredentials", {})
+        user_creds["usernameReference"] = username
+        user_creds["passwordReference"] = password
+        if not all([user_creds["usernameReference"], user_creds["passwordReference"]]):
+            raise RequiredArgumentMissingError(
+                "Please provide username and password reference for UsernamePassword authentication."
+            )
+        auth_props["usernamePasswordCredentials"] = user_creds
+        if auth_props.pop("x509Credentials", None):
+            logger.warning("Previously used certificate reference was removed.")
+    elif auth_mode == AEPAuthModes.anonymous.value and not any([certificate_reference, username, password]):
+        auth_props["mode"] = AEPAuthModes.anonymous.value
+        if auth_props.pop("x509Credentials", None):
+            logger.warning("Previously used certificate reference was removed.")
+        if auth_props.pop("usernamePasswordCredentials", None):
+            logger.warning("Previously used username and password references were removed.")
+    elif any([auth_mode, certificate_reference, username, password]):
+        raise MutuallyExclusiveArgumentError("Invalid combination of authentication mode and parameters.")
+
+    return auth_props
+
+
+def _process_certificates(cert_list: Optional[List[List[str]]] = None) -> List[Dict[str, str]]:
+    """This is for the main create/update endpoint commands"""
+    if not cert_list:
+        return []
+    processed_certs = []
+    for cert in cert_list:
+        parsed_cert = assemble_nargs_to_dict(cert)
+        if set(parsed_cert.keys()) != set(["password", "thumbprint", "secret"]):
```

Comment:
This is probably a nit but just a note - my assumption is that this check is to verify that `password`, `thumbprint`, and `secret` are all provided properties, but this would also fail if extra cert properties are provided in the args right?

Reply:
that is how sets work so yes.

Reply:
I understand how sets work, I'm just curious if it's worth failing the whole command if they add an extra property, rather than just ignoring/logging it.

Reply:
still think it is better to fail since the user can look at their input and fix what went wrong rather than us just assuming it is an oopsie to ignore.